### PR TITLE
Work for release 0.23.x

### PR DESCRIPTION
**Pull request by KBbitsP**
@reshmabidikar @pierre 
Kindly review the changes

### CODE REVIEW

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/json/SubscriptionJson.java [hunk 1]
```diff
@@ -87,6 +88,7 @@ public static class EventSubscriptionJson extends JsonBase {
         private final UUID eventId;
         private final BillingPeriod billingPeriod;
         private final DateTime effectiveDate;
+        private final Date catalogEffectiveDate;
```
Comment (sbrossie) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/json/SubscriptionJson.java:91]: Can you use `DateTime` instead to be consistent, like [here](https://github.com/killbill/killbill/blob/work-for-release-0.23.x/jaxrs/src/main/java/org/killbill/billing/jaxrs/json/InvoiceItemJson.java#L68)

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/json/SubscriptionJson.java [hunk 2]
```diff
@@ -137,6 +141,7 @@ public EventSubscriptionJson(final SubscriptionEvent subscriptionEvent, @Nullabl
             this.eventId = subscriptionEvent.getId();
             this.billingPeriod = billingPeriod;
             this.effectiveDate = subscriptionEvent.getEffectiveDate();
+			this.catalogEffectiveDate = plan != null ? plan.getCatalog().getEffectiveDate() : null;
```
Comment (sbrossie) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/json/SubscriptionJson.java:144]: Indentation is off

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/json/SubscriptionJson.java [hunk 3]
```diff
@@ -174,7 +179,11 @@ public DateTime getEffectiveDate() {
             return effectiveDate;
         }
 
-        public String getPlan() {
+        public Date getCatalogEffectiveDate() {
```
Comment (sbrossie) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/json/SubscriptionJson.java:182]: Indentation: Same here

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/json/SubscriptionJson.java (hunk 1 above)
Comment (reshmabidikar) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/json/SubscriptionJson.java:91]: The `equals` and `hashCode` methods would need to be updated for this additional field

**Review by reshmabidikar — COMMENTED** [review]
Wondering if we need any additional server test (in [TestEntitlement](https://github.com/killbill/killbill/blob/6ae3aa60cd463d9280bc18c77769b2007cb914b9/profiles/killbill/src/test/java/org/killbill/billing/jaxrs/TestEntitlement.java)) to verify that the `catalogEffectiveDate` is returned correctly? Any thoughts @sbrossie?

**Comment by reshmabidikar** [comment]
Wondering if we need any additional server test (in [TestEntitlement](https://github.com/killbill/killbill/blob/6ae3aa60cd463d9280bc18c77769b2007cb914b9/profiles/killbill/src/test/java/org/killbill/billing/jaxrs/TestEntitlement.java)) to verify that the `catalogEffectiveDate` is returned correctly? Any thoughts @sbrossie?

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/json/SubscriptionJson.java (hunk 1 above)
Comment (KBbitsP) [reply jaxrs/src/main/java/org/killbill/billing/jaxrs/json/SubscriptionJson.java:91]: HIi @sbrossie 
For catalog effectiveDate return type is Date as per  returned by "plan.getCatalog().getEffectiveDate() "

Comment (sbrossie) [reply jaxrs/src/main/java/org/killbill/billing/jaxrs/json/SubscriptionJson.java:91]: @KBbitsP Please don;'t resolve comments until we agree on the resolution. I know `plan.getCatalog().getEffectiveDate()` returns a `Date` but all our json  models use `DateTime` so for consistency we need to convert it.

**Review by sbrossie — CHANGES_REQUESTED** [review]

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/json/SubscriptionJson.java (hunk 1 above)
Comment (KBbitsP) [reply jaxrs/src/main/java/org/killbill/billing/jaxrs/json/SubscriptionJson.java]: @sbrossie Converted to DateTime. Please check

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/json/SubscriptionJson.java [hunk 4]
```diff
@@ -20,6 +20,7 @@
 
 import java.math.BigDecimal;
 import java.util.ArrayList;
+import java.util.Date;
```
Comment (sbrossie) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/json/SubscriptionJson.java:23]: Is this 'import' still needed?

**Comment by sbrossie** [comment]
@KBbitsP Can you please add a test that uses the endpoint returning the updated model - see Reshma's [comment](https://github.com/killbill/killbill/pull/1808#pullrequestreview-1200409345) ?

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/json/SubscriptionJson.java [hunk 5]
```diff
@@ -141,7 +141,7 @@ public EventSubscriptionJson(final SubscriptionEvent subscriptionEvent, @Nullabl
             this.eventId = subscriptionEvent.getId();
             this.billingPeriod = billingPeriod;
             this.effectiveDate = subscriptionEvent.getEffectiveDate();
-            this.catalogEffectiveDate = plan != null ? plan.getCatalog().getEffectiveDate() : null;
+            this.catalogEffectiveDate = plan != null ? new DateTime(plan.getCatalog().getEffectiveDate()) : null;
```
Comment (reshmabidikar) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/json/SubscriptionJson.java:144]: @sbrossie, does this conversion from`Date` to `DateTime` look correct or should the `referenceTime`/`TimeAwareContext` be used?

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/json/SubscriptionJson.java (hunk 4 above)
Comment (KBbitsP) [reply jaxrs/src/main/java/org/killbill/billing/jaxrs/json/SubscriptionJson.java]: .Removed extra import.

**Review by sbrossie — APPROVED** [review]

**Pull request merged**